javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -71,6 +73,15 @@
         return copy;
     }
 
+    @SuppressWarnings("unchecked")
+    static <T, U> Object[] copyMap(Function<? super T, ? extends U> mapper, T[] array, int from, int to) {
+        final Object[] copy = array.clone();
+        for (int i = from; i < to; i++) {
+            copy[i] = mapper.apply((T) copy[i]);
```

Comment:
it's faster to write where I read from, instead of creating an empty array and writing the values from the source